### PR TITLE
CustomRPC: show warning when game activity is disabled

### PR DESCRIPTION
**Pull request by nin0-dev**
This PR adds a card at the top of CustomRPC settings that tells the user that their custom RPC won't be visible if their game activity isn't enabled, with a button to quickly enable it.
Here's a recording of the feature in action:
Vesktop_U3CmFe3Xsv
and here's a screenshot of the card itself:
Vesktop_y1csRlCxti

### CODE REVIEW

File: src/plugins/customRPC/index.tsx [hunk 1]
```diff
@@ -386,15 +387,39 @@ async function setRpc(disable?: boolean) {
...
     stop: () => setRpc(true),
     settings,
 
     settingsAboutComponent: () => {
         const activity = useAwaiter(createActivity);
+        const gameActivitySetting = StatusSettingsStores.ShowCurrentGame.useSetting();
+        const [showErrorCard, setShowErrorCard] = useState(!gameActivitySetting);
         return (
             <>
+                <ErrorCard style={{ margin: "20px 0px", padding: "25px 30px", display: !showErrorCard ? "none" : "block" }}>
```
Comment (Vendicated) [inline src/plugins/customRPC/index.tsx:401]: this is not how you conditionally render something in react

```jsx
{showErrorCard && <ErrorCard />}
```

File: src/plugins/customRPC/index.tsx [hunk 2]
```diff
@@ -386,15 +387,39 @@ async function setRpc(disable?: boolean) {
...
-    authors: [Devs.captain, Devs.AutumnVN],
+    authors: [Devs.captain, Devs.AutumnVN, Devs.nin0dev],
     start: setRpc,
     stop: () => setRpc(true),
     settings,
 
     settingsAboutComponent: () => {
         const activity = useAwaiter(createActivity);
+        const gameActivitySetting = StatusSettingsStores.ShowCurrentGame.useSetting();
+        const [showErrorCard, setShowErrorCard] = useState(!gameActivitySetting);
```
Comment (Vendicated) [inline src/plugins/customRPC/index.tsx:398]: this state is not needed. useSetting already is stateful

**Comment by nin0-dev** [comment]
all reviews have been solved

File: src/plugins/customRPC/index.tsx [hunk 3]
```diff
@@ -386,15 +387,41 @@ async function setRpc(disable?: boolean) {
...
     description: "Allows you to set a custom rich presence.",
-    authors: [Devs.captain, Devs.AutumnVN],
+    authors: [Devs.captain, Devs.AutumnVN, Devs.nin0dev],
     start: setRpc,
     stop: () => setRpc(true),
     settings,
 
     settingsAboutComponent: () => {
         const activity = useAwaiter(createActivity);
+        let hideErrorCard = StatusSettingsStores.ShowCurrentGame.useSetting();
```
Comment (Vendicated) [inline src/plugins/customRPC/index.tsx:397]:
```suggestion
        const hideErrorCard = StatusSettingsStores.ShowCurrentGame.useSetting();
```

Comment (Vendicated) [inline src/plugins/customRPC/index.tsx:397]:
```suggestion
        let gameActivityEnabled = StatusSettingsStores.ShowCurrentGame.useSetting();
```

File: src/plugins/customRPC/index.tsx [hunk 4]
```diff
@@ -386,15 +387,41 @@ async function setRpc(disable?: boolean) {
...
+                    <ErrorCard style={{ margin: "20px 0px", padding: "25px 30px" }}>
+                        <Text style={{ fontWeight: "bold", marginBottom: "3px" }}>Notice</Text>
+                        <Text>Game activity isn't enabled, people won't be able to see your custom presence!</Text>
+                        <Button
+                            color={Button.Colors.TRANSPARENT}
+                            style={{ marginTop: "10px" }} onClick={() => {
+                                StatusSettingsStores.ShowCurrentGame.updateSetting(true);
+                                hideErrorCard = true;
+                                const message = "Game activity enabled";
+                                const type = Toasts.Type.SUCCESS;
```
Comment (Vendicated) [inline src/plugins/customRPC/index.tsx:411]: don't create pointless variables

**Pull request merged**